core/src/test/java/com/adobe/aem/commons/assetshare/README.md

```diff
@@ -0,0 +1,149 @@
+# Unit Testing Guide
+
+## Testing Tools
+
+* Unit tests are broken up by the class being tested under `core/src/test/java`.
+* Unit tests are written using the dependency provides in the core/pom.xml:
+    * [JUnit 4.12](https://junit.org/junit4/)
+    * [Mockito 2](https://site.mockito.org/)
+    * [AEM Mocks](http://wcm.io/testing/aem-mock/usage.html)
+
+## Test Cases
+
+Each class being test has a corresponding test case class, sharing the package and filename of the class being test, however the filename is post-fixed with "Test".
+
+For example, to test:
+
+   `core/src/main/java/com.adobe.aem.commons.assetshare.components.impl.HelloWorldImpl.java`
+
+this test is created:
+
+    `core/src/test/java/com.adobe.aem.commons.assetshare.components.impl.HelloWorldImplTest.java`
+
+### Test Methods
+
+Each public method on the class being test should have at least one corresponding test method. If multiple conditions (working/not-working) required testing, multiple test methods should be defined.
+
+Test method naming:
+* Shares the name of method being tested, ex: `getTitle()`
+* Prefixed with `test`, ex: the test method is `testGetTitle()`
+* Optionally post-fixed with an underscore-prefixed, capitalized condition name, ex: `testGetTitle__EmptyTitle()`
+
+
+post-fixed with a clear identifier `_ConditionToTest`
+
+For example, to test in `HelloWorld.java`:
+
+   `String getTitle()`
+
+these test methods created in `HelloWorldImplTest.java`:
+
+    `@Test void testGetTitle()` which tests for the normal, populated title condition
+    `@Test void testGetTitle_EmptyTitle()`, which tests for the condition where the title is missing
+
```

Comment:
I like to name my test methods not only with the name of the method that they are testing, but also the behavior that they are testing.  For example, testGetTitle() would be testGetTitle_ReturnsSetTitle() or testGetTitle_EmptyTitle_ReturnsEmptyString().

Reply:
Lets start with title and condition; I think baking in the expected result into the method name might get super verbose. Ideally, if the tests are well writing it should be quite easy to derive what the expected result should be by reading the failing test.